test/expectations/expectations.go

```diff
@@ -81,22 +81,28 @@ func ExpectApplied(ctx context.Context, c client.Client, objects ...client.Objec
 	}
 }
 
-func ExpectStatusConditions(obj status.Object, conditions ...status.Condition) {
-	GinkgoHelper()
-	objStatus := obj.StatusConditions()
-	for _, cond := range conditions {
-		objCondition := objStatus.Get(cond.Type)
-		Expect(objCondition).ToNot(BeNil())
-		if cond.Status != "" {
-			Expect(objCondition.Status).To(Equal(cond.Status))
-		}
-		if cond.Message != "" {
-			Expect(objCondition.Message).To(Equal(cond.Message))
-		}
-		if cond.Reason != "" {
-			Expect(objCondition.Reason).To(Equal(cond.Reason))
+func EventuallyExpectStatusConditions(ctx context.Context, c client.Client, obj status.Object, timeout time.Duration, conditions ...status.Condition) {
```

Comment:
Let's drop the Eventually -- this is implied by the timeout. Also -- do we want to add the timeout before the object?